From 028a15617a7f2c9172e3ac2d903af0f03010c8b4 Mon Sep 17 00:00:00 2001
From: Christos Zoulas <christos@zoulas.com>
Date: Sun, 26 Apr 2020 17:43:13 +0000
Subject: [PATCH] PR/93: Undo previous: There are dynamic objects (shared
 libraries for example) that don't have interpeters and should not be counted
 as static. We need to figure out what to do with static-pie and gcc-9
 separately.

---
 src/readelf.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/readelf.c src/readelf.c
index 9d1d1ad5..517b3aec 100644
--- src/readelf.c
+++ src/readelf.c
@@ -1646,6 +1646,7 @@ dophn_exec(struct magic_set *ms, int clazz, int swap, int fd, off_t off,
 		switch (xph_type) {
 		case PT_DYNAMIC:
 			doread = 1;
+			linking_style = "dynamically";
 			break;
 		case PT_NOTE:
 			if (sh_num)	/* Did this through section headers */
@@ -1660,7 +1661,6 @@ dophn_exec(struct magic_set *ms, int clazz, int swap, int fd, off_t off,
 			}
 			/*FALLTHROUGH*/
 		case PT_INTERP:
-			linking_style = "dynamically";
 			doread = 1;
 			break;
 		default:
